docs/DeveloperGuide.md

```diff
@@ -1,34 +1,478 @@
+* Storage: Stores the data in an external folder
+* Task: Contains the classes for the different types of tasks.
+* Ui: In charge of interaction with the user.
+
+### Card Package
+The Card package consists of different classes that holds information on the main functionalities of the application,
+and is split into 2 packages, quiz and task, as well as Subject, Topic and Flashcard classes.
+- Quiz Package
+    * Quiz class
+    * Result class
+    * TopicQuiz class
+    * SubjectQuiz class
+- Task Package
+    * Deadline class
+    * Event class
+    * Task class
+    * Todo class
+- Flashcard class
+- Subject class
+- Topic class
+
+Each of the classes in the card package contains:
+* Getters and Setters of its own attributes
+* methods that alter an instance of its own class
+
+#### Quiz package <a name = "quiz"> </a>
+The `Quiz` package holds the necessary classes for the quiz functionality of this application. The following are
+the classes in the package. An abstract`Quiz` class, a `Result` class, a `SubjectQuiz` class and a `TopicQuiz` class.
+
+#### Quiz class <a name = "Qclass"> </a>
+`Quiz` class is an abstract class in the `Quiz` package. It is holds the result of a quiz class and the list of flashcards
+from any subject or topic classes which the user calls the quiz for. Furthermore, it contains a checkAnswer() method the
+checks the answer that the user had given with the correct answer of the quiz. If the user enters the correct answer, the 
+existing score is incremented by one. Else, the contents of the flashcards and the incorrect answer provided by
+the user are transferred to the  `incorrectAnswers` list. Once the user finished the quiz, the application would print the 
+questions that the users did not answer correctly, along the the answer that was provided by the user. 
+
+#### SubjectQuiz <a name = "Qsubject"> </a>
+`SubjectQuiz` class inherits from the `Quiz` class and initiates the quiz. The `startQuiz` method calls for the  `setupQuiz` method checks for the presence
+of topics or flashcards. Else, the application throws the `NoTopicException` for the former, and the 
+`NoFlashcardException` for the latter. If the topics have flashcards, then these are transferred to the 
+`SubjectQuiz` class, while the maximum score of the quiz is set to be the total number of flashcards in the list of flashcards
+present in the `SubjectQuiz` class . 
+
+The application now returns to the `startQuiz` method and the current score of the quiz is set to zero. Subsequently, the
+application begins printing the questions from the flashcards and checks the answer that you provide. If you want to stop the 
+quiz, use the `stop` command. The application will then print out the score that you obtain. If you complete the quiz
+then the application not only prints the score, but it also prints the incorrect answers from the quiz.
+
+The following diagram shows how you can initiate the quiz for a subject.
```

Comment:
It is a bit cramped here to have a bunch of diagrams with just a single line of explanation while there is no numbering or captions of all the diagrams. As a result it might be a bit confusing for readers. You guys can consider put captions/explanations, at least label with diagram is for what.